core/example/longrange/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+add_executable(Direct directsum_example.cpp)
+target_link_libraries(Direct cabanacore)
+add_executable(Ewald ewaldsum_example.cpp)
+target_link_libraries(Ewald cabanacore)
+
+if(Cabana_ENABLE_Cuda)
+  find_package(CUDA REQUIRED)
+  if(CUDA_FOUND)
```

Comment:
With `REQUIRED` above, `CUDA_FOUND` is always true and hence you can drop the condition.